lib/compress.js

```diff
@@ -1273,7 +1273,21 @@ merge(Compressor.prototype, {
         def(AST_Constant, return_false);
         def(AST_This, return_false);
 
+        var pure_regex = /[@#]pure/;
```

Comment:
Or may be just `/^\s*[@#]pure/`, so forcing it to be at the start of the comment?